lisp/forge-commands.el

```diff
@@ -736,17 +744,22 @@ because the source branch has been deleted"))
 
 ;;;###autoload
 (defun forge-checkout-pullreq (pullreq)
-  "Create, configure and checkout a new branch from a pull-request.
-Please see the manual for more information."
+  "Checkout the branch of a PULLREQ."
   (interactive (list (forge-read-pullreq "Checkout pull request" t)))
-  (let ((pullreq (forge-get-pullreq pullreq)))
+  (when-let* ((pullreq (forge-get-pullreq pullreq))
+              (branch (or (if (not (eq (oref pullreq state) 'open))
+                              (magit-ref-p (format "refs/pullreqs/%s"
+                                                   (oref pullreq number)))
+                            (forge--pullreq-branch-active pullreq))
+                          (forge--pullreq-branch-internal pullreq))))
     (magit-checkout
-     (or (if (not (eq (oref pullreq state) 'open))
-             (magit-ref-p (format "refs/pullreqs/%s"
-                                  (oref pullreq number)))
-           (forge--pullreq-branch-active pullreq))
+     (if (or (not (magit-branch-p branch))
+             (eq forge-checkout-reset-existing-branch t)
+             (and (eq forge-checkout-reset-existing-branch 'ask)
+                  (y-or-n-p (format "Reset %s?" branch))))
          (let ((magit-inhibit-refresh t))
-           (forge-branch-pullreq pullreq))))))
+           (forge-branch-pullreq pullreq))
+       branch))))
```

Comment:
I’m not certain I follow everything here, but this looks like it will ask _twice_ if you select `'ask` (once “Reset %s?”, and then “A branch named "%s" already exists … [r][c][a]” when `forge-branch-pullreq` is called) and if you select `t`, then it will still ask once in `forge-branch-pullreq`, no?

I think this change would be better made a bit deeper, in `forge--pullreq-branch-select`, because it’ll improve both `forge-checkout-pullreq` and `forge-checkout-worktree` (I use the latter mostly).

Then an option could be added to the existing prompt to support checking out the existing branch `[u]se existing "%s" branch` (a result you can’t currently get, and in your change is only possible unilaterally).

And finally, the `defcustom` would have options for every case, like `'reset`, `'create`, `'use`, `'abort`, and `nil` (for “ask”). Leaving `nil` as the default so that the behavior doesn’t change for users who don’t set it.